lib/credentials/shared_ini_file_credentials.js

```diff
@@ -48,11 +51,11 @@ AWS.SharedIniFileCredentials = AWS.util.inherit(AWS.Credentials, {
 
     this.filename = options.filename;
     this.profile = options.profile || process.env.AWS_PROFILE || 'default';
-    this.get(function() {});
```

Comment:
I don't think I really understand the issue with removing the `get` in the constructor. Are you concerned about someone directly instantiating SharedIniFileCredentials and not calling `get`? It should get called when the provider chain is resolved right? In what case is `get` not called before retrieving the creds?